[Cocoa] Tell AVFoundation to not log urls in private browsing mode

eric-carlson opened the pull request:
#### b39a8381cc14da138f10ddd6f4d32ed855c0ac1c
<pre>
[Cocoa] Tell AVFoundation to not log urls in private browsing mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=270476">https://bugs.webkit.org/show_bug.cgi?id=270476</a>
<a href="https://rdar.apple.com/118855410">rdar://118855410</a>

Reviewed by Jer Noble.

Add a key to the options dictionary passed to the AVURLAsset constructor when WebKit
is in private browsing mode so it knows to never log urls.

* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.h: Define new constant string.
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::privateBrowsingStateDidChange): Make the code the same on iOS
as on every other platform.

* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::loadWithNextMediaEngine): Update name of instance variable.
(WebCore::MediaPlayer::setPrivateBrowsingMode): Ditto.
* Source/WebCore/platform/graphics/MediaPlayer.h:

* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::createAVAssetForURL): Add new key to options
dictionary when in private browsing mode.

Canonical link: <a href="https://commits.webkit.org/275691@main">https://commits.webkit.org/275691@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aa3014000f762500762eadbda12c1725982ff5bc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42493 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21511 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44888 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45094 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38610 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44800 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24744 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18866 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35180 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43067 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18427 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36575 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16122 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16070 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37624 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/549 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38670 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37948 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46570 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17309 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14250 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41863 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18928 "Built successfully") | | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40471 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/proxy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9514 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18990 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18573 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/platform/graphics/MediaPlayer.h:
```diff
@@ -576,6 +576,7 @@ class WEBCORE_EXPORT MediaPlayer : public MediaPlayerEnums, public ThreadSafeRef
     unsigned videoDecodedByteCount() const;
 
     void setPrivateBrowsingMode(bool);
+    bool privateBrowsingMode() const { return m_privateBrowsingMode; }
```
Minor nit: this bool should verb. Maybe `isInPrivateBrowsingMode()` or just `inPrivate...`.

eric-carlson committed:
[Cocoa] Tell AVFoundation to not log urls in private browsing mode
https://bugs.webkit.org/show_bug.cgi?id=270476
rdar://118855410

Reviewed by Jer Noble.

Add a key to the options dictionary passed to the AVURLAsset constructor when WebKit
is in private browsing mode so it knows to never log urls.

* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.h: Define new constant string.
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::privateBrowsingStateDidChange): Make the code the same on iOS
as on every other platform.

* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::loadWithNextMediaEngine): Update name of instance variable.
(WebCore::MediaPlayer::setPrivateBrowsingMode): Ditto.
* Source/WebCore/platform/graphics/MediaPlayer.h:

* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::createAVAssetForURL): Add new key to options
dictionary when in private browsing mode.

Canonical link: https://commits.webkit.org/275691@main
